wallet/src/libwallet/internal/selection.rs

```diff
@@ -71,6 +71,15 @@ where
 	slate.lock_height = lock_height;
 	slate.fee = fee;
 	let slate_id = slate.id.clone();
+	let mut message = "".to_string();
+	if slate.participant_data.len() > 0 {
+		for data in slate.participant_data.iter() {
+			if message.len() > 0 {
+				message.push_str(&"|".to_string());
```

Comment:
What if the message contains a pipe?

Reply:
I don't know the details of how it all works. What I've tested until now there were always 2 participants. In this case there will always be a senders message only so I don't expect (could be wrong) that there will be two messages (senders and recipients). Could there be 3 participants and thus 3 messages?

About message containing pipe, I didn't account for that since it's not my use case but since message supports any utf8 character there could also be newlines, correct? I'm open to suggestions for a better delimiter.

Alternatively I could even expand the tx slate and store messages separately if there can only be two participants (as in sender's message and recipient's message to respond).

Reply:
Yes, and there will ultimately need multiple participants. A JSON structure would work, the best way to do that would be to serialise a struct that contains an id, public key, signature and the message itself